zerver/views/__init__.py

```diff
@@ -194,6 +201,12 @@ def accounts_register(request):
             # SSO users don't need no passwords
             password = None
 
+        if create_realm and settings.OPEN_REALM_CREATION:
```

Comment:
you can leave off the `settings.OPEN_REALM_CREATION` check here, since you already checked it above.